### ISSUE TITLE
Git w/ Non Standard Port Broken on Poetry 1.0.0b4

### ISSUE THREAD

**Issue by iwoloschin** [sdispater/poetry#1557]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS 10.14.6
- **Poetry version**: 1.0.0b4
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: See Below

## Issue
I'm trying to use a git+ssh URL to a private repo using a non standard port, like this:
```toml
package = {git = "ssh://git@path.to.private.repo.com:1234/repo/project.git"}
```
This leads to an error like this:
```
[ValueError]
Invalid git url ""
```

I think the issue may be here:
https://github.com/sdispater/poetry/blob/master/poetry/vcs/git.py#L16

Specifically, I think this regex is broken:
```python
r"(:?P<port>[\d]+)?"
```
Instead, I think it should look like:
```python
r"(?P<port>:[\d]+)?"
```

If someone doesn't beat me to it I'll likely submit a PR later today or tomorrow.

**Comment by adawalli** [comment]
I can confirm that we just ran into this as well with a VCS dependency on GITLAB, although, we don't even have a port, our URI looked more like

`static-analysis = {git = https://gitlab.internal.com/libraries/static-analysis.git}`

This worked just fine on version b3 and below

**Comment by hongquan** [comment]
Even this Git URL `ssh://git@git.nanochip.vn/elevator/mibielev.git` makes Poetry 1.0.0b4 broken.

**Comment by sdispater** [comment]
The root cause of the issue has been found and a fix is available in #1576.

The issue had nothing to do with the port or the format of the url but a combination of factors that led to this. See the PR for more information.

The fix will be available in the next prerelease.
